api/mysql.proto

```diff
@@ -0,0 +1,73 @@
+syntax = "proto3";
+
+package api;
+
+import "google/api/annotations.proto";
+
+message MySQLNode {
+    reserved 1, 2; // id and type
+
+    string name = 3;
+}
+
+message MySQLService {
+    reserved 1, 2, 3; // id, type, and node_id
+
+    string address = 4;
+    uint32 port = 5;
+    string engine = 6;
+    string engine_version = 7;
+}
+
+message MySQLInstance {
+    MySQLNode node = 1;
+    MySQLService service = 2;
+}
+
+message MySQLListRequest {
+}
+
+message MySQLListResponse {
+    repeated MySQLInstance instances = 1;
+}
+
+message MySQLAddRequest {
+    string address = 1;
+    uint32 port = 2;
+    string username = 3;
+    string password = 4;
+}
+
+message MySQLAddResponse {
+}
+
+message MySQLRemoveRequest {
+    string address = 1;
+    uint32 port = 2;
+}
+
+message MySQLRemoveResponse {
+}
+
+service MySQL {
+
+    rpc List(MySQLListRequest) returns (MySQLListResponse) {
+        option (google.api.http) = {
+            get: "/v0/mysql"
+        };
+    }
+
+    rpc Add(MySQLAddRequest) returns (MySQLAddResponse) {
+        option (google.api.http) = {
+            post: "/v0/mysql"
+            body: "*"
+        };
+    }
+
+    rpc Remove(MySQLRemoveRequest) returns (MySQLRemoveResponse) {
+        option (google.api.http) = {
+            delete: "/v0/mysql"
+            body: "*"
```

Comment:
Let's add id to the URL and drop body. Generator support that natively.

Reply:
ok